lib/modules/commentSortBy.js

```diff
@@ -0,0 +1,52 @@
+/* @flow */
+
+import _ from 'lodash';
+import { $ } from '../vendor';
+import { Module } from '../core/module';
+import { rawLocale } from '../utils/localization';
+import {
+	string,
+	watchForRedditEvents,
+} from '../utils';
+
+export const module: Module<*> = new Module('commentSortBy');
+
+module.beforeLoad = () => {
+	watchForRedditEvents('postModTools', () => {
+		applyToPost();
+	});
+};
+
+module.moduleName = 'commentSortBy';
+module.category = 'commentsCategory';
+module.disabledByDefault = false;
+module.description = 'addCommentSortPrefMenuItemDesc';
+
+export class SortByPrefLink {
+	static defaultLinkElement = (e => () => e().cloneNode(true))(_.once(() => SortByPrefLink.buildTagElement()));
+	static buildTagElement = () => string.html`
+			<span class="RESUserTag">
+				<a
+					class="sortByPrefLink gearIcon"
+					style="opacity: 0.3;"
+					title="Set default sort by preferences"
+					href="https://${rawLocale()}.reddit.com/prefs/#default_comment_sort"
+				>${'\u00A0'/* nbsp */}</a>
+			</span>
+		`;
+
+	static render() {
+		const bttn = SortByPrefLink.defaultLinkElement();
+		if ($('#CommentSort--SortPicker').parent().find('.sortByPrefLink').length === 0) {
```

Comment:
Can you test this inside `applyToPost`, and add a flag to the `postModTools` container instead to verify "this element has been handled"? or you could check the postmodtools `.updates` value.

Reply:
Sounds good! Definitely want to ensure there's only ever one

Reply:
Just curious, when you say "not bulletproof" -- can you describe what you mean, and do you have any suspects what caused it? I'm wondering what unexpected pitfalls we need to learn about when working with the new reddit frontend.

Reply:
Of course, I meant that without it, and contrary to what I thought should happen, I occasionally saw more than one setting gears. This means the id that the update event is receiving is not consistent enough to ensure that there is really only one setting gear. Unfortunately, I could not find the reason for that. Perhaps, and this is really shooting out of my hip, Reddit are reusing DOMs and therefore we're getting the same id in the event...